indico/cli/i18n.py

```diff
@@ -84,25 +91,153 @@ def cli():
 }
 
 
-def find_packages(path, prefix=''):
+def _get_plugin_options(cmd_name, plugin_dir):
+    return {
+        'init_catalog': {
+            'input_file': _get_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+        'extract_messages': {
+            'output_file': _get_messages_pot(plugin_dir),
+            'mapping_file': 'babel.cfg' if os.path.isfile(os.path.join(plugin_dir, 'babel.cfg')) else '../babel.cfg'
+        },
+        'compile_catalog': {
+            'directory': _get_translations_dir(plugin_dir)
+        },
+        'update_catalog': {
+            'input_file': _get_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+
+        # JavaScript
+        'init_catalog_js': {
+            'input_file': _get_js_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+        'extract_messages_js': {
+            'keywords': '$t gettext ngettext:1,2 pgettext:1c,2 npgettext:1c,2,3',
+            'output_file': _get_js_messages_pot(plugin_dir),
+            'mapping_file': 'babel-js.cfg' if os.path.isfile(os.path.join(plugin_dir, 'babel-js.cfg'))
+                            else '../babel-js.cfg'
+        },
+        'update_catalog_js': {
+            'input_file': _get_js_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+
+        # JavaScript / React
+        'init_catalog_react': {
+            'input_file': _get_react_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+        'update_catalog_react': {
+            'input_file': _get_react_messages_pot(plugin_dir),
+            'output_dir': _get_translations_dir(plugin_dir),
+        },
+    }[cmd_name]
+
+
+@contextmanager
+def _chdir(path):
+    cwd = os.getcwd()
+    os.chdir(path)
+    try:
+        yield
+    finally:
+        os.chdir(cwd)
+
+
+def _validate_plugin_dir(plugin_dir):
+    if not os.path.exists(os.path.join(plugin_dir, 'setup.py')):
+        raise click.BadParameter(f'no setup.py found in {plugin_dir}')
+    for dirs in os.listdir(plugin_dir):
+        if (os.path.isdir(os.path.join(plugin_dir, dirs))
+                and 'translations' in os.listdir(os.path.join(plugin_dir, dirs))):
+            break
+    else:
+        raise click.BadParameter(f'no translations folder found in {plugin_dir}')
```

Comment:
If you don't mind `pathlib`, this could be simplified as `Path(plugin_dir).glob('*/*/translations')` (untested though)

```suggestion
    if not list(Path(plugin_dir).glob('*/*/translations')):
        raise click.BadParameter(f'no translations folder found in {plugin_dir}')
```

Reply:
+1 on using pathlib!

Reply:
Yes this works with `.glob('*/translations'))`. Thank you!